### PR TITLE
transform factories into components

### PR DESCRIPTION
**Pull request by esasova**
Transform all factories into React components and support use of factories

Dev Review rules:
- 3 PR max
- at the second feedback, prefer meet instead of github conversation

Definition of Done:
- [ ] Testing
- [ ] No error logs in web console (except iFrame errors)
- [ ] Create or Update documentation (README file for users/integrators)
- [ ] Identify breaking changes with [conventional commits](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index)
- [ ] Check the [test page](https://spaceport.cosmotech.com/confluence/pages/viewpage.action?spaceKey=QA&title=Azure+Sample+Webapp) in order to manually validate the feature
- [ ] Clean code (regroup configuration variable, typos, no console.log...)

### CODE REVIEW

File: src/components/ScenarioParameters/components/ScenarioParametersTabs/ScenarioParameterInput.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  const props = { parameterData, parametersState, setParametersState, editMode, key: parameterData.id };
+  if (parameterVarType.startsWith(DATASET_ID_VARTYPE)) {
+    props.datasets = datasets;
+  }
+  // name property helps distinguish React components from factories
+  if ('name' in varTypeFactory) {
+    return React.createElement(varTypeFactory, props);
+  }
+  return varTypeFactory.create(t, parameterData, parametersState, setParametersState, editMode);
```
Comment (csm-thu) [inline src/components/ScenarioParameters/components/ScenarioParametersTabs/ScenarioParameterInput.js:38]: Could we add a warning after the `if` block to warn users that the "function-shape" factories are now deprecated and won't be supported anymore in a future version ?

File: src/components/ScenarioParameters/components/ScenarioParametersTabs/ScenarioParametersTabsWrapper.js [hunk 2]
```diff
@@ -19,16 +19,26 @@ const ScenarioParametersTabsWrapper = ({
...
+    if ('name' in tabFactory) {
+      parametersGroupMetadata.tab = tabFactory({
+        datasets: datasets,
+        parametersGroupData: parametersGroupMetadata,
+        parametersState: parametersValuesToRender,
+        setParametersState: setParametersValuesToRender,
+        context: context,
+      });
+    } else {
+      parametersGroupMetadata.tab = tabFactory.create(
```
Comment (csm-thu) [inline src/components/ScenarioParameters/components/ScenarioParametersTabs/ScenarioParametersTabsWrapper.js:33]: Could we add another "deprecated" warning here too ?

File: src/utils/scenarioParameters/VarTypesComponentsMapping.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { GENERIC_VAR_TYPES_COMPONENTS_MAPPING } from './generic/VarTypesComponentsMapping';
+import { CUSTOM_VAR_TYPES_COMPONENTS_MAPPING } from './custom/ComponentsMapping';
```
Comment (csm-thu) [inline src/utils/scenarioParameters/VarTypesComponentsMapping.js:5]: Oops, looks like we no longer have consistent names for files in _generic_ and _custom_ folders :sweat_smile: 
We could keep the name `VarTypesComponentsMapping` for both files, to use the same name as the file in the parent folder

Comment (esasova) [reply src/utils/scenarioParameters/VarTypesComponentsMapping.js:5]: Yes, I saw it, it's coming in the next commit :laughing:

**Review by nborde-CSM — APPROVED** [review]
LGTM :+1:

**Review by csm-thu — APPROVED** [review]
LGTM :+1:

**Pull request merged**